thesis-api/main/persistence/term_dao.py

```diff
@@ -0,0 +1,21 @@
+from sqlalchemy.orm import Session
+
+from database.models import TermModel
+from persistence.objects import Term
+
+
+class TermDAO:
+    def __init__(self, db: Session):
+        self.__db = db
+
+    def save_all_of_category(self, terms: list[str], category_id: int):
+        self.__db.add_all([TermModel(name=t, category_id=category_id) for t in terms])
+
+    def get_all_of_category(self, category_id: int) -> list[Term]:
+        models = self.__db.query(TermModel).filter(TermModel.category_id == category_id).all() or []
```

Comment:
I'm not sure if or [] is necessary as the documentation says, all() returns the result of the query as a list. Therefore an empty result is returned as an empty list I guess ...